.travis.yml

```diff
@@ -1,6 +1,9 @@
 language: ruby
 rvm:
   - 2.2.3
+before_script:
+  - sed -f _scripts/baseurl.sed _config.yml > _scripts/_config.yml
+  - mv _scripts/_config.yml _config.yml 
```

Comment:
Excellent feedback, thank you guys!

I have updated this PR to address the latest comments to inline the sed replacement of the baseurl and it now overwrites the existing _config.yml file without a ``mv``.  Thank you for showing me how to do that!

I also added a couple comments into the travis, config, and README files that hopefully will make it explicitly clear what is going on.